### PR TITLE
[ci] Add check to detect likely-nil lua function calls

### LINKED ISSUE #2256: 🔨 Add tooling to verify all Lua binding calls
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have searched existing [issues](https://github.com/LandSandBoat/server/issues) to see if the issue has already been opened, and I have checked the commit log to see if the issue has been resolved since my server was last updated
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)

## Describe the feature

It isn't _that_ hard to scan through all of our Lua scripts and find all instances of `object:functionName`, and verify vs. our bindings; also easily scraped from all instances of `SOL_REGISTER("functionName",`.

While not perfect, and given that objects have different names we can't verify the type, we can verify whether or not a function straight up doesn't exist (or is typo'd).

### PR DESCRIPTION
**Pull request by zach2good**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

## What does this pull request do?

Closes https://github.com/LandSandBoat/server/issues/2256

Hunts down calls to functions on lua usertypes that haven't been properly declared in the script, or as part of a `SOL_REGISTER` call.

## Steps to test these changes

First round:

```
Could not find function match for untargetable (./scripts/globals/nyzul.lua:536:17)
Could not find function match for untargetable (./scripts/globals/nyzul.lua:832:24)
Could not find function match for changeSkin (./scripts/globals/events/harvest_festivals.lua:231:25)
Could not find function match for getAngle (./scripts/globals/interaction/actions/sequence.lua:69:76)
Could not find function match for getRegion (./scripts/globals/items/bottle_of_terroanima.lua:16:16)
Could not find function match for doCuttingCards (./scripts/globals/job_utils/corsair.lua:220:12)
Could not find function match for tryInterruptSpell (./scripts/globals/mobskills/dancing_edge.lua:30:16)
Could not find function match for tryInterruptSpell (./scripts/globals/mobskills/evisceration.lua:30:16)
Could not find function match for tryInterruptSpell (./scripts/globals/mobskills/impulse_drive.lua:38:16)
Could not find function match for tryInterruptSpell (./scripts/globals/mobskills/penta_thrust.lua:31:16)
Could not find function match for tryInterruptSpell (./scripts/globals/mobskills/shadowstitch.lua:38:16)
Could not find function match for tryInterruptSpell (./scripts/globals/mobskills/skewer.lua:38:16)
Could not find function match for tryInterruptSpell (./scripts/globals/mobskills/spiral_hell.lua:30:16)
Could not find function match for tryInterruptSpell (./scripts/globals/mobskills/vorpal_scythe.lua:30:16)
Could not find function match for tryInterruptSpell (./scripts/globals/mobskills/wheeling_thrust.lua:30:16)
Could not find function match for addResist (./scripts/zones/Nyzul_Isle/mobs/Khimaira.lua:19:9)
Could not find function match for AnimationSub (./scripts/zones/The_Garden_of_RuHmet/mobs/Ixzdei_BLM.lua:166:13)
Could not find function match for AnimationSub (./scripts/zones/The_Garden_of_RuHmet/mobs/Ixzdei_BLM.lua:183:9)
Could not find function match for addmob (./scripts/zones/The_Garden_of_RuHmet/mobs/Kfghrah_BLM.lua:17:9)
Could not find function match for addmob (./scripts/zones/The_Garden_of_RuHmet/mobs/Kfghrah_BLM.lua:18:9)
Could not find function match for addmob (./scripts/zones/The_Garden_of_RuHmet/mobs/Kfghrah_WHM.lua:17:9)
Could not find function match for addmob (./scripts/zones/The_Garden_of_RuHmet/mobs/Kfghrah_WHM.lua:18:9)
```

### CODE REVIEW

**Comment by zach2good** [comment]
I'd also be on-board for removing the sections containing `tryInterruptSpell` and raising an issue to note that they should be interrupting

**Comment by zach2good** [comment]
Removed these blocks (since the binding doesn't exist):
```lua
    if
        dmg > 0 and
        skill:getMsg() ~= 31
    then
        target:tryInterruptSpell(mob, info.hitslanded)
    end
    return dmg
```

Rather than adding a block like this _to every mobskill_ or whatever, it should be handled in a relevant global or in cpp

File: scripts/globals/mobskills/impulse_drive.lua [hunk 1]
```diff
@@ -30,14 +30,6 @@ mobskillObject.onMobWeaponSkill = function(target, mob, skill)
...
         end
     end
-
-    if
-        dmg > 0 and
-        skill:getMsg() ~= 31
-    then
-        target:tryInterruptSpell(mob, info.hitslanded)
-    end
-    return dmg
```
Comment (claywar) [inline scripts/globals/mobskills/impulse_drive.lua:40]: Did you intend to remove the return as well?

Comment (zach2good) [reply scripts/globals/mobskills/impulse_drive.lua:40]: Whoops 👁️ 👁️

**Review by claywar — APPROVED** [review]

**Pull request merged**